platform/mbed_rtc_time.h

```diff
@@ -89,6 +89,30 @@ void set_time(time_t t);
  */
 void attach_rtc(time_t (*read_rtc)(void), void (*write_rtc)(time_t), void (*init_rtc)(void), int (*isenabled_rtc)(void));
 
+/** Standard lib retarget, get time since Epoch
+ *
+ * @param timeval   Structure containing time_t secondsa and suseconds_t microseconds. Due to
+ *                  separate target specific RTC implementations only the seconds component is used.
+ *                  It is recommended to use the Mbed OS Time API.
+ * @param timezone  DEPRECATED IN THE STANDARD: This parameter is left in for legacy code. It is
+ *                  not used and will return a warning if a non-NULL value is passed in.
+ * @note Synchronization level: Thread safe
+ *
+ */
+int gettimeofday(struct timeval *tv, struct timezone *tz);
```

Comment:
I'm a bit wary that these structs may not have been defined before this header file - you need to either make sure that the relevant include is done first, or just predeclare the struct tags, otherwise you might be defining a struct tag just in the scope of the declaration.

(Also, tz is so deprecated, `struct timezone` doesn't even exist in POSIX, it's just `void *`, but is this declaration to match what is in the compiler-provided sys/time.h?)

Reply:
Correct I left the timezone parameter in there to match the sys/time.h functions.